src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Internal/HeaderValueProvider.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding.Internal
+{
+    internal class HeaderValueProvider : IValueProvider
+    {
+        private readonly CultureInfo _culture;
+        private readonly bool _isHeaderPresent;
+        private readonly string[] _values;
+
+        public HeaderValueProvider(
+            CultureInfo culture,
+            bool isHeaderPresent,
+            string[] values)
+        {
+            if (values == null)
+            {
+                throw new ArgumentNullException(nameof(values));
+            }
+
+            _culture = culture;
+            _isHeaderPresent = isHeaderPresent;
+            _values = values;
+        }
+
+        /// <inheritdoc />
+        public bool ContainsPrefix(string prefix)
+        {
+            return _isHeaderPresent;
```

Comment:
This is odd and `_isHeaderPresent` shouldn't be necessary. Just use `_values.Length == 0` everywhere.

As-is, `ContainsPrefix(...)` will return `false` and `GetValue(...)` will return a one-element array containing `null` when assigning to a non-collection. The two methods should do sensible things and be consistent.